### DISCUSSION TITLE
Change meow-setup layout without restarting emacs

### DISCUSSION THREAD

**Discussion by jonrubens** [meow-edit/meow#533] [General]
I'm looking for some guidance on how to best change meow commands between using colemak-dh and qwerty. I can manage to type in both but it is hard to also move my meow layout between the two. I'd like the position of all my normal mode keys to match between layouts.

I tried to create a swapping function to just toggle between the two with meow-normal-define-key, but I think I need a way to clear all existing binds first.  What is the right way to approach this? The surefire way I've used is to modify the code and restart emacs, which is not a good fix.

**Comment by eshrh** [comment]
I believe the best way to "clear keybindings" is to wipe the keymap.
```elisp
(setcdr (alist-get 'normal meow-keymap-alist) nil)
```
(note that `(cdr (make-sparse-keymap))` is nil)

Because you want to switch between two keymaps, I would first use meow-normal-define-key to generate your two keymaps, store them in your config, and then your code to swap to a keymap is simply

```elisp
(setcdr (alist-get 'normal meow-keymap-alist) (cdr colemak-normal-keymap))
```
Make sure you set the cdr of the existing keymap, don't set the whole variable. After calling `(meow-normal-define-key)` from default meow, your `meow-normal-state-keymap` will contain the generated keymap.

Because you're just setting the cdr, there's no need to clear it first.
